### DISCUSSION TITLE
Strange Memory beavhiour on Minio Cluster

### DISCUSSION THREAD

**Discussion by Orsopepe** [minio/minio#20710] [Q&A]
Hi guys im quite new on minio but we already running few production clusters with success!

unfortunately a week ago one of our node went oom and start swapping causing the server to hung.
cluster is made by 5 nodes with 90disks 20Tb each, 200GB ram and 32 cores
we noticed a strange beavhiour of the memory; basically minio process using only 11gb but seems all the memory is used (near 200gb)

we already take some actions, like disable the swap (it was left enabled by other collegues in past) and tuned the kernel just only using madvise calls instead always. so THP are no longer used, this doesnt fix the problem.

the thing we cant figure out who is using all this memory when minio is only draining 11gb, we noticed this kind memory consumption on each servers.

i attached some logs to make it more clear.

![immagine (1)](https://github.com/user-attachments/assets/b72832ba-7db8-4413-a459-3fc49fc8791a)
![immagine](https://github.com/user-attachments/assets/f33ee84e-bfe8-4dda-b2a7-8d45b085cf40)

i also attached a meminfo
[root@ltaopeosts06 ~]# cat /proc/meminfo
MemTotal:       197660940 kB
MemFree:        16654972 kB
MemAvailable:   54245236 kB
Buffers:               0 kB
Cached:         20560448 kB
SwapCached:            0 kB
Active:          3714944 kB
Inactive:       20994752 kB
Active(anon):    3625504 kB
Inactive(anon):   626340 kB
Active(file):      89440 kB
Inactive(file): 20368412 kB
Unevictable:      174036 kB
Mlocked:          170976 kB
SwapTotal:             0 kB
SwapFree:              0 kB
Zswap:                 0 kB
Zswapped:              0 kB
Dirty:                 8 kB
Writeback:             0 kB
AnonPages:       4323560 kB
Mapped:           158540 kB
Shmem:             87692 kB
KReclaimable:   19650260 kB
Slab:           22142232 kB
SReclaimable:   19650260 kB
SUnreclaim:      2491972 kB
KernelStack:       31292 kB
PageTables:        28840 kB
SecPageTables:         0 kB
NFS_Unstable:          0 kB
Bounce:                0 kB
WritebackTmp:          0 kB
CommitLimit:    98830468 kB
Committed_AS:   12110896 kB
VmallocTotal:   13743895347199 kB
VmallocUsed:      284524 kB
VmallocChunk:          0 kB
Percpu:            36736 kB
HardwareCorrupted:     0 kB
AnonHugePages:     22528 kB
ShmemHugePages:        0 kB
ShmemPmdMapped:        0 kB
FileHugePages:         0 kB
FilePmdMapped:         0 kB
CmaTotal:              0 kB
CmaFree:               0 kB
Unaccepted:            0 kB
HugePages_Total:       0
HugePages_Free:        0
HugePages_Rsvd:        0
HugePages_Surp:        0
Hugepagesize:       2048 kB
Hugetlb:               0 kB
DirectMap4k:      595968 kB
DirectMap2M:     9492480 kB
DirectMap1G:    192937984 kB

can u guys help me figure out this beavhour and understand better whats happening?

many thanks for your help and time, wish a good Weekend!

Damiano

**Comment by klauspost** [comment]
Since you are clearly beyond an @ home setup, and this will be be a deeper investigation than just debugging some commandline option, reach out to `sales@min.io` on support options.
